src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,121 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double INCREASING_SIZE = 1.5;
+    private static final int START_POSITION = 0;
+    private static final int STEP_SIZE = 1;
+    private static final String INDEX_OUT_OF_BOUNDS_MESSAGE =
+            "Index %d out of bounds for size: %d";
+    private int size;
+    private Object[] array = new Object[DEFAULT_CAPACITY];
+
     @Override
     public void add(T value) {
-
+        if (size < array.length) {
+            array[size] = value;
+            size++;
+        } else {
+            increaseArray();
+            add(value);
+        }
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index >= 0 && index <= size) {
+            if (size >= array.length) {
+                increaseArray();
+            }
+            shiftArrayWithAddingElement(value, index);
+        }
+        checkIndex(index);
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
```

Comment:
Foreach is more suitable here

Reply:
For this I have to implement iterator, should I do it?